<launch>

    <!-- RViz user args -->
    <arg name="rviz"             default="true"/>
    <arg name="rviz_path"  	     default="$(find cameras)/rviz/d435i_cables"/>
	<arg name="pub_base_link_tf" default="true"/>
    <!-- Image format -->
    <arg name="image_width"     default="640"/>
    <arg name="image_height"    default="480"/>
    <arg name="steps_cable"     default="10"/>

    <!-- 3D pose estimation with d435i camera launch -->
	<include file="$(find cameras)/launch/d435i/pose3D_estimator_d435i.launch" >
		<arg name="rviz" 				value="true"/>
        <arg name="rviz_path"  	        value="$(arg rviz_path)"/>
    	<arg name="pub_base_link_tf"    value="$(arg pub_base_link_tf)"/>
    </include>

    <!-- FASTDLO Detection server  -->
    <node pkg="cables_detection" type="fastdlo_server.py" name="fastdlo_server">
        <param name="image_width"   type = "int" value="$(arg image_width)"/>
        <param name="image_height"  type = "int" value="$(arg image_height)"/>
        <param name="steps_cable"   type = "int" value="$(arg steps_cable)"/>
    </node>

</launch>